### PR TITLE
Archive module preview 2 & 3

### PR DESCRIPTION
**Pull request by ayousuf23**
# PR Summary

This PR includes code for the preview 2 release of the Microsoft.PowerShell.Archive v2.0.0 module.

This PR is a rewrite of the Expand-Archive cmdlet in C# (previously, it was written in PowerShell Script). It has support for creating, updating, and overwriting archives. As of now, only zip and zip64 are supported (the .NET APIs for zip and zip64 are the same).

The PR has a -Format parameter which can used to specify the format of the archive. As of now, -Format has only one option: -Format Zip. When additional formats are supported by the module in preview 3, the -Format parameter will be more useful by providing options for tar and compressed tar.

To overwrite the destination path of the archive or items in the archive, use the -WriteMode Overwrite parameter.

For more information regarding behavior of the parameters, please see the RFC https://github.com/PowerShell/Microsoft.PowerShell.Archive/pull/117.

## PR Context

To keep the PowerShell/Microsoft.PowerShell.Archive repository up to date with preview 2 code.

### CODE REVIEW

**Comment by SydneyhSmith** [comment]
Looks like this repo might be missing a change log-- that would be a good thing to add

**Comment by ayousuf23** [comment]
@SydneyhSmith I added the changelog

File: CHANGELOG.md [hunk 1]
```diff
@@ -0,0 +1,20 @@
+# Changelog
+
+## 2.0.1-preview2
+
+- Rewrite `Expand-Archive` cmdlet in C#
+- Added `-Format` parameter to `Expand-Archive`
+- Added `-WriteMode` parameter to `Expand-Archive`
+- Added support for zip64
+- Fixed a bug where the entry names of files in a directory would not be correct when compressing an archive
+- `Compress-Archive` skips writing an entry to an archive if an error occurs while doing so
```
Comment (daxian-dbw) [inline CHANGELOG.md:10]: The preview.2 change log has duplicate items from preview.1. Can you please clean them up to only contain the new improvements in preview2?

Comment (ayousuf23) [reply CHANGELOG.md]: @daxian-dbw I updated the changelog.

**Pull request merged**